### PR TITLE
Use env `RUNBOOK_URL_TEMPLATE` for the runbooks URL template

### PR DESCRIPTION
**Pull request by assafad**
**What this PR does / why we need it**:
When `RUNBOOK_URL_TEMPLATE` env variable is set, its value will be used as the runbooks URL template. Otherwise, keep using the default runbook URL template `https://kubevirt.io/monitoring/runbooks/%s`. 
This allows using different runbooks versions, in accordance with the installation.

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
https://issues.redhat.com/browse/CNV-14383
```

**Release note**:

```release-note
Use env RUNURL_TEMPLATE as the runbooks URL template
```

### CODE REVIEW

**Comment by assafad** [comment]
/hold
let's wait for https://github.com/kubevirt/monitoring/pull/164 to be merged

**Comment by assafad** [comment]
@nunnatsa, @sradco Can you please review?

**Review by nunnatsa — COMMENTED** [review]
Good work! 

Added a few comments.

/ok-to-test

File: controllers/alerts/alerts.go [hunk 1]
```diff
@@ -33,17 +36,31 @@ const (
...
+	template string
+}
 
-	outOfBandUpdateRunbookUrl             = fmt.Sprintf(runbookUrlTemplate, outOfBandUpdateAlert)
-	unsafeModificationRunbookUrl          = fmt.Sprintf(runbookUrlTemplate, unsafeModificationAlert)
-	installationNotCompletedRunbookUrl    = fmt.Sprintf(runbookUrlTemplate, installationNotCompletedAlert)
-	nodeStatusMaxImagesExceededRunbookUrl = fmt.Sprintf(openshiftRunbookUrlTemplate, nodeStatusMaxImagesExceededAlert)
-)
+func newRunbookCreator() *runbookCreator {
+	runbookURLTemplate, exists := os.LookupEnv("RUNURL_TEMPLATE")
```
[the remark this answers is not in the document]
Comment (assafad) [reply controllers/alerts/alerts.go]: done

[the remark this answers is not in the document]
Comment (assafad) [reply controllers/alerts/alerts.go]: done

File: controllers/alerts/metrics_test.go [hunk 2]
```diff
@@ -306,6 +307,41 @@ var _ = Describe("alert tests", func() {
...
+				for _, rule := range group.Rules {
+					if rule.Alert != "" {
+						if rule.Annotations["runbook_url"] != "" {
+							Expect(rule.Annotations["runbook_url"]).To(Equal(fmt.Sprintf(desiredRunbookURLTemplate, rule.Alert)))
+						}
+					}
+				}
+			}
+
+			os.Unsetenv("RUNURL_TEMPLATE")
```
[the remark this answers is not in the document]
Comment (assafad) [reply controllers/alerts/metrics_test.go]: done

File: tests/func-tests/monitoring_test.go [hunk 3]
```diff
@@ -77,11 +77,11 @@ var _ = Describe("[crit:high][vendor:cnv-qe@redhat.com][level:system]Monitoring"
 			for _, rule := range group.Rules {
 				if rule.Alert != "" {
 					Expect(rule.Annotations).To(HaveKeyWithValue("summary", Not(BeEmpty())),
-						fmt.Sprintf("%s summary is missing or empty", rule.Alert))
+						fmt.Sprintf("%s summary annotation is missing or empty", rule.Alert))
```
Comment (nunnatsa) [inline tests/func-tests/monitoring_test.go:80]: You don't need `fmt.Sprintf`. The `To` method can get format and parameters.
```suggestion
				Expect(rule.Annotations).To(HaveKeyWithValue("summary", Not(BeEmpty())), "%s summary annotation is missing or empty", rule.Alert)
```

File: tests/func-tests/monitoring_test.go [hunk 4]
```diff
@@ -77,11 +77,11 @@ var _ = Describe("[crit:high][vendor:cnv-qe@redhat.com][level:system]Monitoring"
...
+					Expect(rule.Annotations).To(HaveKeyWithValue("runbook_url", Not(BeEmpty())),
+						fmt.Sprintf("%s runbook_url annotation is missing or empty", rule.Alert))
```
Comment (nunnatsa) [inline tests/func-tests/monitoring_test.go:82]: same

**Pull request merged**